napari/_qt/layer_controls/_tests/test_qt_layer_controls.py

```diff
@@ -1,31 +1,333 @@
+            for idx in range(qcombobox_count):
+                previous_qcombobox_text = qcombobox.currentText()
+                qcombobox.setCurrentIndex(idx)
+                # If a value for the QComboBox is an invalid selection check if
+                # it fallbacks to the previous value
+                captured = capsys.readouterr()
+                if captured.err:
+                    assert qcombobox.currentText() == previous_qcombobox_text
+            qcombobox.setCurrentIndex(qcombobox_initial_idx)
+
+    # check QAbstractSpinBox by changing value with `setValue` from minimum value to maximum
+    for qspinbox in ctrl.findChildren(QAbstractSpinBox):
+        qspinbox_initial_value = qspinbox.value()
+        qspinbox_min = qspinbox.minimum()
+        qspinbox_max = qspinbox.maximum()
+        if isinstance(qspinbox_min, float):
+            if np.isinf(qspinbox_max):
+                qspinbox_max = sys.float_info.max
+            value_range = np.linspace(qspinbox_min, qspinbox_max)
+        else:
+            # use + 1 to include maximum value
+            value_range = range(qspinbox_min, qspinbox_max + 1)
+            try:
+                value_range_length = len(value_range)
+            except OverflowError:
+                # range too big for even trying to get how big it is.
+                value_range_length = 100
+                value_range = [
+                    random.randrange(qspinbox_min, qspinbox_max)
+                    for _ in range(value_range_length)
+                ]
+                value_range.append(qspinbox_max)
+            if value_range_length > 100:
+                # prevent iterating over a big range of values
+                random.seed(0)
+                value_range = random.sample(value_range, 100)
+                value_range = np.insert(value_range, 0, qspinbox_min)
+                value_range = np.append(value_range, qspinbox_max - 1)
+        for value in value_range:
+            qspinbox.setValue(value)
+            # capture any output done to sys.stdout or sys.stderr.
+            captured = capsys.readouterr()
+            assert not captured.out
+            if captured.err:
+                # since an error was found check if it is associated with a known issue still open
+                expected_errors = [
+                    "MemoryError: Unable to allocate",  # See https://github.com/napari/napari/issues/5798
+                    "ValueError: array is too big; `arr.size * arr.dtype.itemsize` is larger than the maximum possible size.",  # See https://github.com/napari/napari/issues/5798
+                    "ValueError: Maximum allowed dimension exceeded",  # See https://github.com/napari/napari/issues/5798
+                    "IndexError: index ",  # See https://github.com/napari/napari/issues/4864
+                    "RuntimeWarning: overflow encountered",  # See https://github.com/napari/napari/issues/4864
+                ]
```

Comment:
Can you make a note in each of these issues that any PR to close them should also remove these lines from this test? (You might have to do it after this PR is merged, in order to link to the line(s) directly)

Reply:
Thanks for the feedback! Will do :+1: